### ISSUE TITLE
var-naming rule can suggest invalid identifiers

### ISSUE THREAD

**Issue by jonathangjertsen** [mgechev/revive#977]
**Describe the bug**

I have a program in which there is a concept of "struct" which does not refer to go structs. `struct` is not a valid identifier, so I use `struct_`. This triggers the `var-naming` check, which says `don't use underscores in Go names; var struct_ should be struct`.

**To Reproduce**

1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with the following flags & no configuration file:

```shell
# flags

revive ./... | grep struct_
```

**Expected behavior**

If the result of removing the underscore is an invalid identifier, `var-naming` should either

1. Allow leading or trailing underscore
2. Suggest a valid identifier

**Logs**

N/A

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.3 LTS
 - Version of Go: 1.22

**Additional context**

This issue seems related to #613, but regardless of whether one thinks underscores in this usecase, the check should not suggest an invalid name

**Comment by chavacava** [comment]
Hi @, thanks for reporting the issue. 

Using _ in identifiers is not idiomatic in Go, that is why the rule complains on `record_`

Sure, the proposed fix (`record`) is not valid because it results to be a reserved word of the language. It could be possible to avoid these kinds of incorrect recommendations by checking against the list of reserved words.  
In these cases, it could be tricky to propose an alternative name. Thus, the rule could just signal the naming problem without proposing a fix.

IMO, adding these checks isn't worth the performance penalties.

In your case, you could just use `strukt` to avoid the names clash.

**Comment by chavacava** [comment]
I will close the issue for now, and if it turns out to be impacting a larger number of users, I'll reopen it.

**Comment by WGH-** [comment]
Just for the record, `type_` seems acceptable in Go compiler code base: https://github.com/search?q=org%3Agolang+%2F%5Cttype_%5Cs%2B%2F&type=code
